### ISSUE TITLE
Magic Create keeps breaking as we evolve the API

### ISSUE THREAD

**Issue by aldonline** [decoupled/magic#8]
As we evolve the magic APIs, [Magic Create](https://dev.decoupled.com/magic-create ) keeps breaking.
This happens because the code is generated dynamically (given the options entered by the user), so we don't type check all possible results.

Proposed fix: When compiling the website, we need a way to generate all permutations of possible options automatically and check them with the compiler against the latest types for "@decoupled/magic".

<img width="807" alt="Screen Shot 2020-01-30 at 1 05 09 PM" src="https://user-images.githubusercontent.com/154884/73466988-6d212980-4361-11ea-977c-efeb27ab3ffd.png">

**Comment by aldonline** [comment]
Lol. There are more than 130k valid combinations of settings right now.
I need to add some extra restrictions to rule out combinations that don't make sense, like a project with a database and nothing else.
```
{
    fabricui: false,
    materialui: false,
    spa: false,
    spa_webWorker: false,
    electron: false,
    electron_title: 'My Desktop App',
    electron_main: false,
    electron_webWorker: false,
    lambda: false,
    cmd: false,
    cmd_pjson_cmd: 'my-cmd',
    cmd_pjson_name: 'my-command-cli',
    lib: false,
    lib_pjson_name: 'lib',
    lib_genDocs: false,
    cron: false,
    cron_expr: 'xx',
    ssc: false,
    useReactRouter: false,
    useReachRouter: false,
    orm: false,
    orm_forms: false,
    orm_crud: false,
    cms: false,
    mysql: false,
    postgresql: false,
    docker: false,
    netlify: false,
    netlify_identity: false,
    netlify_triggers: false
  },
  ... 130972 more items
```

**Comment by aldonline** [comment]
Or... we can differentiate between monotonic settings (that just add, never alter what another setting did) and those that are, effectively, alternatives. I need to untangle the code a bit but that might work.

**Comment by aldonline** [comment]
* Solved in v0.0.22 - (versions are shared across NPM library, VSCode extension, and website. You can find more info [here](https://dev.decoupled.com/version) )
* The Interactive Project Builder is now called ["Magic Create"](https://dev.decoupled.com/magic-create) and is located here. It works with the CLI
* Only 30 or so combinations are tested.
